git/README.md

```diff
@@ -181,3 +181,71 @@ There are 4 additional parameters in addition to the ones mentioned above for th
 [git-tree]: https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
 [tide]: https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/README.md
 [zuul-ci]: https://zuul-ci.org/docs/zuul/discussion/gating.html
+
+
+## `git-rebase`
+
+This [task](../git/git-rebase.yaml) will `rebase` the branch based
+on the user input. Before rebase, if `squashing` of the commits is required,
+then it can be done by providing the `squash count` i.e number of commits to squash.
+
+### Workspaces
+
+* **source**: A workspace that contains the fetched the git repository.
+
+### Parameters
+
+* **SQUASH_REQUIRED**: A boolean value which specifies if 
+squashing commits is required or not before rebase (_default:_ `true`).
+* **SQUASH_COUNT**: Number of commits to squash in the branch.
+(_only required if squashing needs to be performed_).
+* **COMMIT_MSG**: Commit message to add in commit after the
+ squashing is done (_only required if squashing needs to be performed_).
+* **GIT_USER_NAME**: Git user name to use for rebase (_required_).
+* **GIT_USER_EMAIL**: Git user email to use for rebase (_required_).
+* **PULL_REMOTE_NAME**: Git remote name from which we have to pull 
+and rebase.(_default_: `origin`).
+* **PULL_REMOTE_URL**: Git remote URL from which we have to 
+pull and rebase (_required to set remote).
+* **PULL_BRANCH_NAME**: Git branch name from which we have to 
+pull and rebase (_required_).
+* **PUSH_REMOTE_NAME**: Git remote name to push after rebase(_default_: `origin`).
+* **PUSH_REMOTE_URL**:  Git remote URL to push after rebase(_required_).
+* **PUSH_BRANCH_NAME**: Git branch name to push after rebase (_required_).
+
+
+### Results
+
+* **commit**: The precise commit SHA after the rebase.
+
+### Usage
+
+This task needs authentication to git in order to push after the rebase.
+This can be done via `secrets` and `service account`. Refer [this](../git/git-rebase-example) 
+example to create secret and service account for the git authentication.
+
+[This](../git/git-rebase-example) pipeline has two tasks, one to clone a 
```

Comment:
1. Can we specify name of the pipeline to refer instead of `This` ?
2. I would have used `git-rebase` instead of `git-rebase-example`

Reply:
Yes, we can do this :+1: